Payload_Type/hannibal/hannibal/agent_code/Hannibal/linux_makefile

```diff
@@ -63,25 +65,31 @@ $(BIN_DIR)/$(PROJECT).exe: $(ASM_OBJ_FILES) $(OBJ_FILES)
 	@rm -f "$(BIN_DIR)\$(PROJECT).exe"
 	@rm -rf bin/obj/*.o 
 
-$(OBJ_DIR)/%.o: $(SRC_DIR)/%.c
-	@echo "[+] Compiling $? -> $@"
-	@ $(CC_X64) -o $@ -c  $? $(CFLAGS) $(LDFLAGS)
+$(OBJ_DIR)/%.o: $(SRC_DIR)/%.c $(DEP_DIR)/%.d | $(DEP_DIR)
+	@echo "[+] Compiling $< -> $@"
+	@ $(CC_X64) -o $@ -c  $< $(DEPFLAGS) $(CFLAGS) $(LDFLAGS)
 
 $(ASM_OBJ_FILES): $(OBJ_DIR)/%.o: asm/x64/%.asm
 	@echo "[+] Assembling $? -> $@"
 	@ nasm -f win64 $? -o $@
 
+$(DEP_DIR): ; @mkdir -p $@
+$(DEP_FILES):
+
+-include $(DEP_FILES)
+
 #---------- Utility ----------#
 
 clean:
 	@rm -rf bin/obj/*.o 
 	@rm -rf bin/obj/*.s
 	@rm -rf bin/obj/*.ii
+	@rm -rf $(DEP_DIR)
 	@rm -rf bin/*.bin
 	@rm -rf bin/*.exe
 
 print:
 	@echo "SRC_FILES: $(SRC_FILES)"
 	@echo "OBJ_FILES: $(OBJ_FILES)"
 	@echo "ASM_FILES: $(ASM_FILES)"
-	@echo "ASM_OBJ_FILES": $(ASM_OBJ_FILES)"
```

Comment:
I'm assuming you meant to put `@echo "DEP_FILES: $(DEPFILES)" `here. 

I also need to remove the extra ": `@echo "ASM_OBJ_FILES" <---- : $(ASM_OBJ_FILES)"`

Reply:
Doh, I meant `@echo "DEP_FILES: $(DEP_FILES)" `

Reply:
I added the `@echo "DEP_FILES: $(DEP_FILES)"` line in the `print` target here 3b4672d6e194285bc25340b849b0db1df981ebc1.